src/topology/topological_fiber_bundle.lean

```diff
@@ -979,3 +988,214 @@ begin
 end
 
 end topological_fiber_bundle_core
+
+section topological_fiber_prebundle
+
+variables (F) {Z : Type*} [topological_space B] [topological_space F] {proj : Z → B}
+
+/-- This structure contains the information left for a local trivialization if the total
+space has not been given a topology, but we have a topology on both the fiber and the base space.
+Through the construction `topological_fiber_prebundle F proj` it will be possible to promote a
+`prebundle_trivialization F proj` to a `bundle_trivialization F proj`. -/
+@[nolint has_inhabited_instance]
+structure prebundle_trivialization (proj : Z → B) extends local_equiv Z (B × F) :=
+(open_target   : is_open target)
+(base_set      : set B)
+(open_base_set : is_open base_set)
+(source_eq     : source = proj ⁻¹' base_set)
+(target_eq     : target = set.prod base_set univ)
+(proj_to_fun   : ∀ p ∈ source, (to_local_equiv p).1 = proj p)
+
+instance : has_coe_to_fun (prebundle_trivialization F proj) := ⟨_, λ e, e.to_fun⟩
+
+variable {F}
+
+@[simp, mfld_simps] lemma prebundle_trivialization.coe_coe (e : prebundle_trivialization F proj) :
+  ⇑e.to_local_equiv = e := rfl
+
+/-- Composition of inverse and coercion from the subtype of the target. -/
+def prebundle_trivialization.set_symm (e : prebundle_trivialization F proj) : e.target → Z :=
+set.restrict e.to_local_equiv.symm e.target
+
+lemma prebundle_trivialization.mem_source (e : prebundle_trivialization F proj)
+  {x : Z} : x ∈ e.source ↔ proj x ∈ e.base_set :=
+by rw [e.source_eq, mem_preimage]
```

Comment:
I wonder if a `bundle_trivialization` should extend `prebundle_trivialization` and all this lemmas should be proven only once.

Reply:
Yes, I think this is a very good idea. You would still need to state the lemmas for `bundle_trivialization`s (to make them available for dot notation and rewrites) but all their proofs would become one-liners. This would mean moving everything on prebundle trivializations higher in the file, of course.

Reply:
Actually this is very problematic, because it would involve putting `set_option old_structure_cmd true` before the definition of local_homeomorph, and this breaks so much stuff. Do you want to have a look or should we give up?

Reply:
(I don't mean trying to do the modification in the file on fiber bundles, but just to add the option before `local_homeomorph` and fix the errors that arise! The rest I can do it myself!)

Reply:
You shouldn't need to use old style structures. You can do it as follows: don't change the current definitions of `prebundle_trivialization` and `bundle_trivialization`, but add a definition `bundle_trivialization.to_prebundle_trivialization`. Then all the lemmas for bundle trivializations should follow readily from their analogue for prebundle trivialization. If you don't see how to do that, tell me and I'll have a go at it.

Reply:
Ok sure I can do that

Reply:
I reorganized the code but I was not able to make lean recognize the old proofs for the new lemmas. This does not matter much I guess as all proofs are already made up by one line.